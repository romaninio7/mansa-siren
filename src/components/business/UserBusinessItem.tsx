import * as React from 'react';
import { IUserBusinessItem } from 'interfaces';

const UserBusinessItem = ({
  businessValue,
  businessData,
}: IUserBusinessItem): JSX.Element => {
  return (
    <div className='item'>
      <div className={`ui label large ${businessData[0]}`}>
        <i className={`id icon ${businessData[1]}`}></i>
        <span className='business-name'>{businessData[2]}: </span>
      </div>
      <span className={businessData[3]}>{businessValue}</span>
    </div>
  );
};

export default UserBusinessItem;
