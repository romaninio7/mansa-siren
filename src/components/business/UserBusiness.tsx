import * as React from 'react';

//Components
import Loader from 'components/helpers/Loader';
import Error from 'components/helpers/Error';
import UserBusinessItem from 'components/business/UserBusinessItem';

//Functions & API
import useFetch from 'hooks/useFetch';
import sirenAPI from 'api/siren';

//Data & Structure
import * as consts from 'consts';
import { IBusinessData } from 'interfaces';

const UserBusiness = (): JSX.Element => {
  const { fetchedData, error, fetching } = useFetch(sirenAPI, consts.sirenUser);

  const businessData: IBusinessData = {
    siret: fetchedData?.data.unite_legale.etablissements[0].siret,
    startDate: fetchedData?.data.unite_legale.etablissement_siege.date_debut,
    adresse: fetchedData?.data.unite_legale.etablissements[0].geo_adresse,
  };

  if (error) {
    return <Error>{error}</Error>;
  }

  if (fetching) {
    return <Loader linesNumber={6} />;
  }

  const businessListRender = (): JSX.Element[] => {
    return Object.values(businessData).map((item, index) => {
      return (
        <UserBusinessItem
          businessValue={item}
          businessData={consts.businessData[index]}
          key={index}
        />
      );
    });
  };

  return <div className='ui divided list large'>{businessListRender()}</div>;
};

export default UserBusiness;
