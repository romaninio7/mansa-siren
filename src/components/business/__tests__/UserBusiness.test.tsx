import * as React from 'react';
import UserBusiness from 'components/business/UserBusiness';
import UserBusinessItem from 'components/business/UserBusinessItem';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

let wrapper;

describe('UserBusiness', () => {
  beforeEach(() => {
    act(() => {
      wrapper = mount(<UserBusiness />);
    });
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('has a valid shanpshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders 3 UserBusinessItem', () => {
    expect(wrapper.find(UserBusinessItem).length).toEqual(3);
  });
});
