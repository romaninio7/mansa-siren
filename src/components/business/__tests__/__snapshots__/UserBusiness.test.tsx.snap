// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBusiness has a valid shanpshot 1`] = `
<UserBusiness>
  <div
    className="ui divided list large"
  >
    <UserBusinessItem
      businessData={
        Array [
          "blue",
          "briefcase",
          "Siret",
          "siret",
        ]
      }
      key="0"
    >
      <div
        className="item"
      >
        <div
          className="ui label large blue"
        >
          <i
            className="id icon briefcase"
          />
          <span
            className="business-name"
          >
            Siret
            : 
          </span>
        </div>
        <span
          className="siret"
        />
      </div>
    </UserBusinessItem>
    <UserBusinessItem
      businessData={
        Array [
          "teal",
          "calendar alternate",
          "Start date",
          "startDate",
        ]
      }
      key="1"
    >
      <div
        className="item"
      >
        <div
          className="ui label large teal"
        >
          <i
            className="id icon calendar alternate"
          />
          <span
            className="business-name"
          >
            Start date
            : 
          </span>
        </div>
        <span
          className="startDate"
        />
      </div>
    </UserBusinessItem>
    <UserBusinessItem
      businessData={
        Array [
          "pink",
          "map marker alternate",
          "Address",
          "address",
        ]
      }
      key="2"
    >
      <div
        className="item"
      >
        <div
          className="ui label large pink"
        >
          <i
            className="id icon map marker alternate"
          />
          <span
            className="business-name"
          >
            Address
            : 
          </span>
        </div>
        <span
          className="address"
        />
      </div>
    </UserBusinessItem>
  </div>
</UserBusiness>
`;
